design-proposals/passt-feature-gate-iface-api-deprecation.md

```diff
@@ -0,0 +1,161 @@
+      Existing VMs that use Passt interface API won't manage to re-start.
+    
+      Users who like to use Passt binding will have to follow the network binding plugin API:
+      1. Install Passt binding CNI (make sure the binary available on all nodes).
+      2. Create and register Passt binding network-attachment-definition.
+      3. Register Passt binding sidecar image.
+         
+   2. Remove Passt binding implementation from Kubevirt core.
+   
+   3. Remove Passt feature gate.
+
+## API Examples
+### VM with Passt interface using the retiring Passt interface API:
+1. Enable Passt feature:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+      - Passt
+  ...
+```
+
+2. Specify Passt interface in the VM spec template:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: myvm
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+          - name: blue
+            passt: {}
+      networks:
+      - name: blue
+        pod: {}
+  ...
+```
+
+### VM with Passt interface using network binding API:
+
+1. Passt network binding plugin network-attachment-definition should be created, see the following example:
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: kubevirt-passt-binding
+spec:
+  config: '{
+            "cniVersion": "0.3.1",
+            "type": "kubevirt-passt-binding"
+  }'
+```
+
+2. Register Passt network binding network-attachment-definition and sidecar image, by specifying them in Kubevirt configuration:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+      - NetworkBindingPlugins
+    network:
+      Binding:
+        passt:
+          sidecarImage: quay.io/kubevirt/network-passt-binding:v1.0.0
+          networkAttachmentDefinition: kubevirt-passt-binding
+  ...
+```
+
+3. Specify the registered Passt network binding plugin in VM spec template:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: myvm
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+          - name: blue
+            binding:
+              name: passt
+      networks:
+      - name: blue
+        pod: {}
+  ...
+```
+
+
+## Scalability
+
+## Update/Rollback Compatibility
+In Kubevirt v1.3.0: 
+- Existing running VMs that use Passt interface API will keep running with no disruption.
+- Stopped VMs or new ones, will not manage to start and require using the network binding API.
+- Passt doesnt support migration.
+
+## Functional Testing Approach
+Kubevirt's Passt network e2e tests shall spin up VMs with Passt interface using network binding plugin API.
+
+## Implementation Phases
```

Comment:
Please ref to the section above for the plan, or just repeat it here in a very short form.